r/R/metadata.R

```diff
@@ -72,7 +72,10 @@ apply_arrow_r_metadata <- function(x, r_metadata) {
         # we cannot apply this row-level metadata, since the order of the rows is
         # not guaranteed to be the same, so don't even try, but warn what's going on
         trace <- trace_back()
-        in_dplyr_collect <- any(map_lgl(trace$calls, function(x) {
+        # TODO: remove the option munging once rlang > 0.4.11 is released
+        old_opts <- options(warnPartialMatchDollar = FALSE)
+        on.exit(options(old_opts))
+        in_dplyr_collect <- any(map_lgl(trace$call, function(x) {
```

Comment:
We could also do this and avoid messing with options:

```suggestion
        in_dplyr_collect <- any(map_lgl(trace$calls %||% trace$call, function(x) {
```